pkg/strategy/streaming.go

```diff
@@ -15,84 +17,235 @@ import (
+
+	// these are the in-container paths, available to the Jaeger containers (ingester/collector)
+	clusterCAPath := fmt.Sprintf("/var/run/secrets/%s-cluster-ca", jaeger.Name)
+	clientCertPath := fmt.Sprintf("/var/run/secrets/%s", ku.Name)
+
+	// this is the volume containing the client TLS details, like the cert and key
+	kuVolume := corev1.Volume{
+		Name: ku.Name,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: ku.Name,
+			},
+		},
+	}
+	// this is the volume containing the CA cluster cert
+	kuCAVolume := corev1.Volume{
+		Name: fmt.Sprintf("%s-cluster-ca", jaeger.Name), // the cluster name is the jaeger name
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: fmt.Sprintf("%s-cluster-ca-cert", jaeger.Name),
+			},
+		},
+	}
+	jaeger.Spec.Volumes = replaceVolume(jaeger.Spec.Volumes, kuVolume, kuCAVolume)
+
+	// and finally, the mount paths to have the secrets in the container file system
+	kuVolumeMount := corev1.VolumeMount{
+		Name:      ku.Name,
+		MountPath: clientCertPath,
+	}
+	kuCAVolumeMount := corev1.VolumeMount{
+		Name:      fmt.Sprintf("%s-cluster-ca", jaeger.Name), // the cluster name is the jaeger name
+		MountPath: clusterCAPath,
+	}
+	jaeger.Spec.VolumeMounts = replaceVolumeMount(jaeger.Spec.VolumeMounts, kuVolumeMount, kuCAVolumeMount)
+
+	brokers := fmt.Sprintf("%s-kafka-bootstrap.kafka.svc.cluster.local:9093", k.Name)
+
+	collectorOpts := asOptionsMap(jaeger.Spec.Collector.Options.Map())
+	ingesterOpts := asOptionsMap(jaeger.Spec.Ingester.Options.Map())
+
+	collectorOpts["kafka.producer.brokers"] = brokers
+	collectorOpts["kafka.producer.authentication"] = "tls"
+	collectorOpts["kafka.producer.tls.ca"] = fmt.Sprintf("%s/ca.crt", clusterCAPath)
+	collectorOpts["kafka.producer.tls.cert"] = fmt.Sprintf("%s/user.crt", clientCertPath)
+	collectorOpts["kafka.producer.tls.key"] = fmt.Sprintf("%s/user.key", clientCertPath)
+	if _, found := collectorOpts["kafka.producer.topic"]; !found {
+		collectorOpts["kafka.producer.topic"] = "jaeger-spans"
+	}
+
+	ingesterOpts["kafka.consumer.brokers"] = brokers
+	ingesterOpts["kafka.consumer.authentication"] = "tls"
+	ingesterOpts["kafka.consumer.tls.ca"] = fmt.Sprintf("%s/ca.crt", clusterCAPath)
+	ingesterOpts["kafka.consumer.tls.cert"] = fmt.Sprintf("%s/user.crt", clientCertPath)
+	ingesterOpts["kafka.consumer.tls.key"] = fmt.Sprintf("%s/user.key", clientCertPath)
+	if _, found := collectorOpts["kafka.consumer.topic"]; !found {
+		ingesterOpts["kafka.consumer.topic"] = "jaeger-spans"
+	}
+
+	jaeger.Spec.Collector.Options = v1.NewOptions(collectorOpts)
+	jaeger.Spec.Ingester.Options = v1.NewOptions(ingesterOpts)
+
+	return manifest
+}
+
+func asOptionsMap(in map[string]string) map[string]interface{} {
+	out := make(map[string]interface{})
+	for k, v := range in {
+		out[k] = v
+	}
+	return out
+}
+
+func replaceVolume(volumes []corev1.Volume, name ...corev1.Volume) []corev1.Volume {
```

Comment:
Same for these - I believe there is already utility functions for handling overridden volumes/mounts etc.